src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/RandomPartitioningQueryAndScanMapper.java

```diff
@@ -7,121 +7,91 @@
-import java.util.ArrayList;
 import java.util.Collection;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Optional;
-import java.util.Set;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.function.BiFunction;
-import javax.annotation.Nullable;
+import java.util.function.Function;
 
 /**
  * {@linkn QueryAndScanMapper} implementation for shared tables using random partitioning.
  *
  * @author msgroi
  */
-class RandomPartitioningQueryAndScanMapper implements QueryAndScanMapper {
+class RandomPartitioningQueryAndScanMapper extends AbstractQueryAndScanMapper {
 
     private static final String VALUE_PLACEHOLDER = ":___value___";
-    private static final Map<ComparisonOperator, BiFunction<String, String, String>>
-        FIELD_AND_VALUE_TO_EXPRESSION_STRINGS = new ImmutableMap
-        .Builder<ComparisonOperator, BiFunction<String, String, String>>()
-        .put(EQ, (field, value) -> field + " = " + value)
-        .put(GT, (field, value) -> field + " > " + value)
-        .put(GE, (field, value) -> field + " >= " + value)
-        .put(LT, (field, value) -> field + " < " + value)
-        .put(LE, (field, value) -> field + " <= " + value)
-        .build();
-    private static final Set<ComparisonOperator> SUPPORTED_COMPARISON_OPERATORS = FIELD_AND_VALUE_TO_EXPRESSION_STRINGS
-        .keySet();
 
     private final FieldMapper fieldMapper;
-    private final RandomPartitioningTableMapping tableMapping;
-
-    RandomPartitioningQueryAndScanMapper(RandomPartitioningTableMapping tableMapping, FieldMapper fieldMapper) {
+    private final List<FieldMapping> tablePrimaryKeyFieldMappings;
+    private final Map<DynamoSecondaryIndex, List<FieldMapping>> indexPrimaryKeyFieldMappings;
+
+    RandomPartitioningQueryAndScanMapper(DynamoTableDescription virtualTable,
+                                         Function<String, RequestIndex> indexLookup,
+                                         ConditionMapper conditionMapper,
+                                         ItemMapper itemMapper,
+                                         FieldMapper fieldMapper,
+                                         List<FieldMapping> tablePrimaryKeyFieldMappings,
+                                         Map<DynamoSecondaryIndex, List<FieldMapping>> indexPrimaryKeyFieldMappings) {
+        super(virtualTable, indexLookup, conditionMapper, itemMapper);
         this.fieldMapper = fieldMapper;
-        this.tableMapping = tableMapping;
+        this.tablePrimaryKeyFieldMappings = tablePrimaryKeyFieldMappings;
+        this.indexPrimaryKeyFieldMappings = indexPrimaryKeyFieldMappings;
     }
 
     @Override
-    public void apply(QueryRequest queryRequest) {
-        validateQueryRequest(queryRequest);
-        apply(new QueryRequestWrapper(queryRequest), true);
-    }
+    ScanResult executeScanInternal(AmazonDynamoDB amazonDynamoDB, ScanRequest scanRequest) {
```

Comment:
Logic for random partitioning scan is basically still the same. Changes are due to refactoring.